cli_client/python/timesketch_cli_client/cli.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2021 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Timesketch CLI client."""
+
+import sys
+import click
+
+from requests.exceptions import ConnectionError as RequestConnectionError
+
+from timesketch_api_client import config as timesketch_config
+
+from timesketch_cli_client.commands import analyze
+from timesketch_cli_client.commands import config
+from timesketch_cli_client.commands import importer
+from timesketch_cli_client.commands import search
+from timesketch_cli_client.commands import sketch as sketch_command
+from timesketch_cli_client.commands import timelines
+
+from .definitions import DEFAULT_OUTPUT_FORMAT
+from .version import get_version
+
+
+class TimesketchCli(object):
+    """Timesketch CLI state object.
+
+    Attributes:
+        sketch_from_flag: Sketch ID if provided by flag
+        config_assistant: Instance of ConfigAssistant
+    """
+    def __init__(self, api_client=None, sketch_from_flag=None, conf_file=None):
+        """Initialize the state object.
+
+        Args:
+            sketch_from_flag: Sketch ID if provided by flag.
+        """
+        self.api = api_client
+        self.sketch_from_flag = sketch_from_flag
+
+        if not api_client:
+            try:
+                self.api = timesketch_config.get_client(load_cli_config=True)
+                if not self.api:
+                    raise RequestConnectionError
+            except RequestConnectionError:
+                click.echo('ERROR: Cannot connect to the Timesketch server.')
+                sys.exit(1)
+
+        self.config_assistant = timesketch_config.ConfigAssistant()
+        if conf_file:
```

Comment:
well, technically this is not needed...

```python
self.config_assistant.load_config_file(config_file_path=conf_file, load_cli_config=True)
```

should suffice... by default `conf_file` is set to None, and hence will be passed on like that to the function...you should perhaps instead of default to `None` to default to `''` (empty string) since that is what the library expects to get (a string that is)... but you don't really need this `if` condition, since having an empty value means the default location will be picked

Reply:
done